### PR TITLE
freeze and band

### PR DESCRIPTION
**Pull request by constd**

### CODE REVIEW

File: packages/generation/src/generation/recipe/gan_recipe.py [hunk 1]
```diff
@@ -56,14 +56,15 @@ def training_step(
...
-                clean_mix, generated_mix.detach()
+                clean_mix, generated_mix
             )
 
             discriminator_loss = self.loss.discriminator(discriminator_output)
 
-            discriminator_optimizer.zero_grad()
-            self.manual_backward(discriminator_loss["loss"], retain_graph=True)
-            discriminator_optimizer.step()
+            if self.global_step >= 10000:
```
Comment (cjacoby) [inline packages/generation/src/generation/recipe/gan_recipe.py]: I know this draft, but can we make this configurable / enableable?

Comment (constd) [reply packages/generation/src/generation/recipe/gan_recipe.py]: worse than draft, this hack and yes, we could/should

Comment (cjacoby) [reply packages/generation/src/generation/recipe/gan_recipe.py]: A funny note / observation:

global steps are per optimizer step; so with three datasets, it shows up in the graph as 10000 / 3 😅

File: packages/generation/src/generation/recipe/gan_recipe.py [hunk 2]
```diff
@@ -56,14 +58,15 @@ def training_step(
...
-                clean_mix, generated_mix.detach()
+                clean_mix, generated_mix
             )
 
             discriminator_loss = self.loss.discriminator(discriminator_output)
 
-            discriminator_optimizer.zero_grad()
-            self.manual_backward(discriminator_loss["loss"], retain_graph=True)
-            discriminator_optimizer.step()
+            if self.global_step >= self.num_frozen_steps:
```
Comment (cjacoby) [inline packages/generation/src/generation/recipe/gan_recipe.py:66]: I was looking at the loss graphs, and they didn't seem quite right, so I went back and compared this to how it's done in bigvgan:

https://github.com/NVIDIA/BigVGAN/blob/main/train.py#L486

I their implementation (different than here), the `discriminator_optimizer.zero_grad()` is _always_ called, but backwards and step are optionally called. 

I think the way you have it here we may be implicitly stepping the discriminator through the generator because it's not zeroing the gradients.

File: packages/generation/src/generation/recipe/gan_recipe.py (hunk 1 above)
Comment (constd) [reply packages/generation/src/generation/recipe/gan_recipe.py]: we should put that in a comment in the docstring
